src/FunderMaps.WebApi/Controllers/Report/InquirySampleController.cs

```diff
@@ -13,83 +13,149 @@
 
-            var result = await _mapper.MapAsync<IList<InquirySampleDto>, InquirySample>(_inquiryUseCase.GetAllSampleAsync(pagination.Navigation))
-                .ConfigureAwait(false);
+            // Map.
+            var output = await _mapper.MapAsync<IList<InquirySampleDto>, InquirySample>(inquirySampleList);
 
-            return Ok(result);
+            // Return.
+            return Ok(output);
         }
 
+        // POST: api/inquiry/{id}/sample/{id}
+        /// <summary>
+        ///     Create inquiry sample.
+        /// </summary>
         [HttpPost]
         public async Task<IActionResult> CreateAsync(int inquiryId, [FromBody] InquirySampleDto input)
         {
-            if (input == null)
+            // Map.
+            var inquirySample = _mapper.Map<InquirySample>(input);
+            inquirySample.Inquiry = inquiryId;
+
+            // Act.
+            // FUTURE: Too much logic
+            var inquiry = await _inquiryRepository.GetByIdAsync(inquirySample.Inquiry);
+            if (!inquiry.State.AllowWrite)
             {
-                throw new ArgumentNullException(nameof(input));
+                throw new EntityReadOnlyException();
             }
 
-            var inquirySample = _mapper.Map<InquirySample>(input);
-            inquirySample.Inquiry = inquiryId;
+            inquirySample = await _inquirySampleRepository.AddGetAsync(inquirySample);
 
-            inquirySample = await _inquiryUseCase.CreateSampleAsync(inquirySample).ConfigureAwait(false);
+            inquiry.State.TransitionToPending();
+            await _inquiryRepository.SetAuditStatusAsync(inquiry.Id, inquiry);
 
-            return Ok(_mapper.Map<InquirySampleDto>(inquirySample));
+            // Map.
+            var output = _mapper.Map<InquirySampleDto>(inquirySample);
+
+            // Return.
+            return Ok(output);
         }
 
+        // PUT: api/inquiry/{id}/sample/{id}
+        /// <summary>
+        ///     Update inquiry by id.
+        /// </summary>
         [HttpPut("{id:int}")]
         public async Task<IActionResult> UpdateAsync(int inquiryId, int id, [FromBody] InquirySampleDto input)
         {
-            if (input == null)
-            {
-                throw new ArgumentNullException(nameof(input));
-            }
-
+            // Map.
             var inquirySample = _mapper.Map<InquirySample>(input);
             inquirySample.Id = id;
             inquirySample.Inquiry = inquiryId;
 
-            await _inquiryUseCase.UpdateSampleAsync(inquirySample).ConfigureAwait(false);
+            // Act.
+            var inquiry = await _inquiryRepository.GetByIdAsync(inquirySample.Inquiry);
+            if (!inquiry.State.AllowWrite)
+            {
+                throw new EntityReadOnlyException();
+            }
+
+            await _inquirySampleRepository.UpdateAsync(inquirySample);
+
+            inquiry.State.TransitionToPending();
```

Comment:
I suggest adding this to the remarks of the call, since this is something that will be overlooked easily in the future.

Reply:
Added issue #343